kuma/static/js/components/user-signup/auth-modal.js

```diff
@@ -0,0 +1,41 @@
+(function() {
+    'use strict';
+
+    var authModalContainer = document.getElementById('auth-modal');
+
+    if (
+        !authModalContainer ||
+        (window.matchMedia && window.matchMedia('(max-width:47.9385em)').matches)
+    ) {
+        return;
+    }
```

Comment:
@Gregoor I am thinking now that this piece(the core of it) can(should?) be moved to `modal.js` so it is available to all modal implementations in the same manner. Should we want to change the logic for it, we can do it in one place and it will be inherited by all modal. I am thinking something like:

```
...
shouldShowModal: function(modal) {
  if (
    !modal ||
    (window.matchMedia && window.matchMedia('(max-width:47.9385em)').matches)
  ) {
      return false;
  } else {
      return true;
  }
}
...
```

Then inside a modal implementation such as this:

```
var modal = document.getElementById('auth-modal');

if (!window.mdn.modalDialog.shouldShowModal(modal)) {
  return;
}
```

Thoughts?

Reply:
Yes, I think that makes a lot of sense given that probably all modals should share that behavior!